### ISSUE TITLE
unsupported by backend contract: module initializers

### ISSUE THREAD

**Issue by RiRi114** [llvm/torch-mlir#2527]
I need convert my model to linalg, but I meet a bug. I simplify the model, the bug target to nn.BatchNorm1d.

**Simplified model is :**
```
import torch
import torch.nn as nn
import torch_mlir

class BNTest(nn.Module):
    def __init__(self):
        super().__init__()
        self.batchNorm_1 = nn.BatchNorm1d(32)

    def forward(self, X):
        deep_out = self.batchNorm_1(X)
        return deep_out
    
input =  torch.randn(50,32)
bntest = BNTest()
print(bntest(input))
module = torch_mlir.compile(bntest, input, output_type="LINALG_ON_TENSORS")
with open('bntest.mlir', "w", encoding="utf-8") as outf:
    outf.write(str(module))
```

**the error is :**
```
python exception: Failure while executing pass pipeline:
error: unknown: unsupported by backend contract: module initializers
note: unknown: see current operation: "torch.initialize.global_slots"(%6) <{slotSymNames = [@batchNorm_1.num_batches_tracked]}> : (!torch.tensor<[],si64>) -> ()
note: unknown: this is likely due to InlineGlobalSlots being unable to inline a global slot
```

I have used newest version torch-mlir. How to solve this error? thanks for answering me!

**Comment by ramiro050** [comment]
Hi @RiRi114, PyTorch `nn.Module`s run with `trining=True` by default. Some modules during training mode have internal state that gets mutated on every call. This is the case for `nn.batchNorm1d`. This module has a parameter called `num_batches_tracked` that gets incremented by 1 on every call when `training=True`. Because torch-mlir does not support keeping track of module state, the lowering fails.

In order to not have it fail, you need to do

```diff
  input =  torch.randn(50,32)
  bntest = BNTest()
+ bntest.train(False)
  print(bntest(input))
  module = torch_mlir.compile(bntest, input, output_type="LINALG_ON_TENSORS")
  with open('bntest.mlir', "w", encoding="utf-8") as outf:
      outf.write(str(module))
```

**Comment by RiRi114** [comment]
Thanks for your reply. I get it.
